spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  describe "GET #index" do
+    subject(:index) { get :index }
+
+    it { is_expected.to be_successful }
+  end
+
+  describe "GET #show" do
+    subject(:show) { get :show, params: { id: article.id } }
+    let(:article) { create(:article) }
+
+    it { is_expected.to be_successful }
+  end
+
+  describe "POST #create" do
+  context "when the article is valid" do
+    let(:valid_attributes) { { title: "Very valid title", body: "Very very valid value", published: true } } 
+  subject(:post_create) {post :create, params: { article: valid_attributes}}
+  it "creates new articles" do 
+    expect{subject}.to change(Article, :count).from(0).to(1)
+ end
+it "redirect to article" do 
+    subject
+    expect(response).to redirect_to (articles_path)
+end
+end
+context "when the article is invalid" do
+  let(:invalid_attributes) { { title: "a", body: "This is a valid body." } } # Assuming the title is invalid
+
+  subject(:create_article) { post :create, params: { article: invalid_attributes } }
+
+  it "does not create a new article" do
+    expect { create_article }.not_to change(Article, :count)
+  end
+
+  it "renders the new template" do
+    create_article
+    expect(response).to have_http_status(:unprocessable_entity)  
+    expect(response).to render_template(:new) 
```

Comment:
developerzy Railsów radzą odchodzić od testowania renderów, zapewne stąd u Ciebie ten dodany gem. w tym wypadku należy sprawdzić tylko czy nie zmieniła się liczba artykułów oraz status http, i to wystarczy :)